src/index.js

```diff
@@ -1,53 +1,58 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 
-class Typographer extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { statefulGenerator: this.generator(props.text) };
-    }
-
-    componentDidMount() {
-        this.typingTimer = setInterval(
-            () => this.tick(), this.props.speed
-        );
-    }
-
-    componentWillUnmount() {
-        clearInterval(this.typingTimer);
-    }
-
-    render() {
-        var statefulGenerator = this.state.statefulGenerator;
-        return (<span> {statefulGenerator.text} </span>);
-    }
-
-    tick() {
-        this.setState({
-            statefulGenerator: this.state.statefulGenerator.next()
-        });
-    }
-
-    generator(text) {
-        return this.generatorHelper(text, 0);
-    }
-
-    generatorHelper(text, currentIndex) {
+export default class Typographer extends React.PureComponent {
+	constructor(props) {
+		super(props);
+		this.state = {
+			statefulGenerator: this.generator(props.children || props.text)
+		};
+	}
+
+	componentDidMount() {
+		this.typingTimer = setInterval(() => this.tick(), this.props.speed);
+	}
+
+	componentWillUnmount() {
+		clearInterval(this.typingTimer);
+	}
+
+	tick() {
+    const {statefulGenerator} = this.state;
+		this.setState({
+			statefulGenerator: statefulGenerator.next()
+		});
+	}
+
+	generator(text) {
+		return this.generatorHelper(text, 0);
+	}
+
+	generatorHelper(text, currentIndex) {
+    switch (typeof text) {
+      case "object":
+        return this.generatorHelper(text.map((t) => t.props.children).join(''), currentIndex)
```

Comment:
I feel like it would be better if this went in its own function. Also would be nice if we have more than just one level of nesting. Not sure if this would work since I didn't test it but I was thinking that we could do something like this:


```
getAllChildrenOfComponent(com, acc) {
    if (!com.props.children) {
        return com;
    } else {
        const children = com.props.children;
        return children.map( (child) => getAllChildrenOfComponent(child, [...acc, child])); 
    }
}
```

```
convertToText(root) {
    return getAllChildrenOfComponent(root, []).join('');
}
```

and then after we do `convertToText`, we can call the `generatorHelper` function. That way, that function doesn't need to care about whether what was passed in was an object or string. Logic in `generatorHelper` becomes

```
	generatorHelper(text, currentIndex) {
        return {
          text: text.substring(0, currentIndex + 1),
          next: () => this.generatorHelper(text, currentIndex + 1),
          finished: currentIndex < text.length
        };
	}
```